addons/website_sale/static/src/js/website_sale_service.js

```diff
@@ -0,0 +1,541 @@
+import {
+    ComboConfiguratorDialog
+} from '@sale/js/combo_configurator_dialog/combo_configurator_dialog';
+import { ProductCombo } from '@sale/js/models/product_combo';
+import {
+    ProductConfiguratorDialog
+} from '@sale/js/product_configurator_dialog/product_configurator_dialog';
+import { serializeComboItem } from '@sale/js/sale_utils';
+import { browser } from '@web/core/browser/browser';
+import { serializeDateTime } from '@web/core/l10n/dates';
+import { _t } from '@web/core/l10n/translation';
+import { rpc } from '@web/core/network/rpc';
+import { registry } from '@web/core/registry';
+import { session } from '@web/session';
+
+const { DateTime } = luxon;
+
+/**
+ * @typedef {Object} CustomAttributeValues
+ * @property {Number} custom_product_template_attribute_value_id
+ * @property {String} custom_value
+ */
+
+/**
+ * Manages product addition via the {@link addToCart} function.
+ *
+ * This function handles the process of adding products to the cart, including:
+ * - Opening configurators if needed;
+ * - Updating the cart with the selected products;
+ * - Updating the cart count in the navbar;
+ * - Notifying the customer of successful additions;
+ * - Track the added products.
+ *
+ * Override this class to implement additional checks or
+ * provide relevant information when adding a product to the cart.
+ */
+export class WebsiteSaleService {
```

Comment:
Why not addToCartService?

Reply:
This would make a weird call: `this.addToCart.addToCart()` or `this.call('addToCart').addToCart()` instead of `this.websiteSale.addToCart()` or `this.call('websiteSale').addToCart()`

Reply:
Then, whatever you deem suitable in the range `.*cart.*`. E.g., `this.cartService.add()` or `this.cartService.addProduct()`.

Don't pick names to suit a codebase that doesn't exist yet, may never exist, or in a way that you didn't anticipate.